src/constitutive/TACSBladeStiffenedShellConstitutive.cpp

```diff
@@ -1890,21 +1742,32 @@ TacsScalar TACSBladeStiffenedShellConstitutive::computeStiffenerFailure(
   TACSOrthotropicPly* ply = this->stiffenerPly;
 
   // Compute the strain state at the tip of the stiffener
-  TacsScalar zTipOffset = -(this->stiffenerHeight + this->stiffenerThick) -
-                          this->computeStiffenerCentroidHeight();
-  TacsScalar tipStrain[3];
-  memset(tipStrain, 0, 3 * sizeof(TacsScalar));
-  tipStrain[0] = stiffenerStrain[0] + zTipOffset * stiffenerStrain[2];
+  TacsScalar zCentroid = this->computeStiffenerCentroidHeight();
```

Comment:
Right, it's because the stiffener is assumed to be oriented opposite from the shell normal, so the offset of the stiffener centroid from the shell mid-plane is negative. I've changed it now so that `computeStiffenerCentroidHeight` and `computeStiffenerCentroidHeightSens` treat the height as a positive value and updated all the places they're used to account for the change.